src/index.ts

```diff
@@ -36,6 +36,7 @@ function focusable(el: Target): boolean {
 function visible(el: Target): boolean {
   return (
     !el.hidden &&
+    !el.closest('details:not([open])') &&
```

Comment:
The first summary in a details would actually be shown so we probably need something like:

```suggestion
    !(el instanceof HTMLSummaryElement && el.closest('details:not([open])')) &&
```

But we need to make sure to test this.